api/src/main/java/jakarta/data/repository/Pageable.java

```diff
@@ -142,6 +140,21 @@ public static Pageable ofSize(int maxPageSize) {
     @Override
     public boolean equals(Object o);
 
+    /**
+     * Returns the keyset values which are the starting point for
+     * keyset pagination.
+     *
+     * @return the keyset values; <code>null</code> if using offset pagination.
+     */
+    public Cursor getCursor();
```

Comment:
I'm not a big fan of API that returns null. 
We have other options, such as an `Optional` or checking if it makes sense to put Pageable with this responsibility.

Reply:
but all the other parts of the API return `null` and `Optional` hasn't been used anywhere. If we move to `Optional` we should be consistent

Reply:
I agree, if a value might return null it should return an Optional IMHO.

Reply:
I'm willing to switch this to `Optional` if that's what everyone wants. Personally I didn't see the point because this method will always return a valid cursor when in the CURSOR_NEXT or CURSOR_PREVIOUS mode, and always `null` when in the OFFSET mode, so no one should ever use the `Optional` for anything besides `Optional.get()` which is just extra code path because you already know what you are getting from the mode.

Reply:
@otaviojava @graemerocher I added `Optional` as requested under https://github.com/jakartaee/data/pull/73/commits/356b656622d775a65ac308e1979560849e4e2127
Please review.
@keilw you indicated after discussion on today's Jakarta Data call that you are okay with this approach now. If you have any other review comments let me know.

Reply:
Sorry for the inconvenience - just after posting the above comment I found this pull had significant merge conflicts with #74 and I needed to rebase onto latest to resolve, which required updating the commit link above.